src/helpers/useUndo.ts

```diff
@@ -98,17 +110,22 @@ export const useUndo = ({
   nodes,
   edges,
   disabled,
+  initialHistory,
   maxHistory = 20,
   onUndoRedo
 }: UndoProps): UndoResult => {
   const [canUndo, setCanUndo] = useState<boolean>(false);
   const [canRedo, setCanRedo] = useState<boolean>(false);
 
-  const manager = useRef<Undoo>(
-    new Undoo({
-      maxLength: maxHistory
-    })
-  );
+  const undoo = new Undoo({
```

Comment:
This has to be inside the ref otherwise its recreated every render

Reply:
I need to reuse the `undoo` to use `undoo.import()` below, how do I keep track of the `undoo` reference if it's inside the ref?

Would you mind fixing this? I don't really see how it can be done.

Reply:
I tried doing the following, but it didn't pass the tests:

```ts
  const manager = useRef<Undoo>(
    new Undoo({
      maxLength: maxHistory
    })
  );

  useEffect(() => {
    if (Array.isArray(initialHistory)) {
      manager?.current?.undoo?.import(initialHistory);
    }
  }, [initialHistory, manager]);
```

Reply:
You shouldn't need to pass a ref in the dep array.

Reply:
Maybe, (but then eslint complains about it), but it doesn't fix the implementation.

I don't understand how to do that, might be simpler if you take a look yourself, if you have the time.

Reply:
What did the tests complain about?

Reply:
The last test didn't pass anymore. It basically broke the feature. Here's the output:

![image](https://user-images.githubusercontent.com/3807458/109979451-343f4b00-7cff-11eb-8607-a1f93fa24b16.png)

It's this test, which initializes with an existing history, but "undoing" doesn't restore it anymore:

```ts
  test('should allow to load an initial history', () => {
    // Renders the component, make "screen" available
    render(
      <UndoRedoMockComponent
        initialHistory={[{ nodes: [{ id: '1' }, { id: '2' }], edges: [] }]}
      />
    );

    expect(
      screen.getByTestId('output'),
      'Should be initialised with 0 nodes and 0 edges'
    ).toHaveTextContent('There are 0 nodes and 0 edges.');

    fireEvent.click(screen.getByText('Undo'));
    expect(
      screen.getByTestId('output'),
      'Undoing should go back to the previous state of the history, which contains 2 nodes and 0 edges'
    ).toHaveTextContent('There are 2 nodes and 0 edges.');
  });
```